src/main/java/donk/Parser.java

```diff
@@ -0,0 +1,142 @@
+            ui.bye();
+            return;
+        } else if (fullCommand.equals("list")) {
+            ui.listTasks(tasks);
+        } else if (inputArray[0].equals("mark") && inputArray[1].matches("\\d+")) {
+            int index = Integer.parseInt(inputArray[1]) - 1;
+            Task temp = tasks.getTask(index);
+            temp.markDone();
+            ui.markedDone(temp);
+        } else if (inputArray[0].equals("unmark") && inputArray[1].matches("\\d+")) {
+            int index = Integer.parseInt(inputArray[1]) - 1;
+            Task temp = tasks.getTask(index);
+            temp.unmarkDone();
+            ui.unmarkedDone(temp);
+        } else if (command.equals("delete")) {
+            if (inputArray.length < 2) {
+                throw new IllegalArgumentException ("Please provide the index of the task to delete");
+            }
+            if (!Parser.validNum(inputArray[1])) {
+                throw new IllegalArgumentException("Please provide a valid index");
+            }
+            int index = Integer.parseInt(inputArray[1]) - 1;
+            tasks.remove(index);
+            ui.delete(tasks.getTask(index), tasks.size());
+        }else if (command.equals("todo") || command.equals("deadline") || command.equals("event")) {
+
+            Task t;
+            if (inputArray[0].equals("todo")) {
+                if (inputArray.length < 2 || fullCommand.length() < 5) {
+                    throw new TodoException("Bruh todo task cannot be empty.\nExample:\n    todo homework");
+                }
+                t = new ToDo(fullCommand.substring(5));
+
+                tasks.add(t);
+            } else if (inputArray[0].equals("deadline")) {
+                String[] split = fullCommand.split("/by");
+                if (split.length < 2) {
+                    ui.invalidFormat(TaskType.DEADLINE);
+                    return;
+                }
+                String bef = split[0].substring(9);
+                String aft = split[1];
+                t = new Deadline(bef, aft.strip());
+                tasks.add(t);
+
+            } else {
+                String[] split1 = fullCommand.split("/start");
+                if (split1.length < 2) {
+
+                    return;
+                }
+                String[] split2 = split1[1].split("/end");
+                if (split1.length < 2 || split2.length < 2) {
+                    ui.invalidFormat(TaskType.EVENT);
+                    return;
+                }
+                String start = split2[0];
+                String end = split2[1];
+                String description = split1[0].substring(6);
+                t = new Event(description, start.strip(), end.strip());
+                tasks.add(t);
+
+            }
+            System.out.println("    Got it. I've added this task:");
+            System.out.println("        " + t.toString());
+            System.out.println("    You now have " + tasks.size() + " tasks");
```

Comment:
Since we have an existing `Ui.java` class for printing to the console, should we move into `Ui.java` for consistency?